pkg/server/container_create_windows.go

```diff
@@ -459,9 +459,44 @@ func (c *criService) generateContainerSpec(id string, sandboxID string, sandboxP
 		for _, group := range securityContext.GetSupplementalGroups() {
 			g.AddProcessAdditionalGid(uint32(group))
 		}
-		setOCINamespaces(&g, securityContext.GetNamespaceOptions(), sandboxPid)
 		g.SetProcessNoNewPrivileges(securityContext.GetNoNewPrivs())
+
+		if c.config.DisableCgroup {
+			g.SetLinuxCgroupsPath("")
+		} else {
+			setOCILinuxResourceCgroup(&g, config.GetLinux().GetResources())
+			// LCOW does not support custom cgroup parents. Set it to empty here
+			// and let the GCS place it as the parent of the sandbox.
+			g.SetLinuxCgroupsPath("")
+		}
+		// TODO: JTERRY75 - c.config.RestrictOOMScoreAdj needs to be proxied to LCOW
+		if err := setOCILinuxResourceOOMScoreAdj(&g, config.GetLinux().GetResources(), false); err != nil {
+			return nil, err
+		}
+
+		setOCINamespaces(&g, securityContext.GetNamespaceOptions(), sandboxPid)
 	} else {
+		resources := config.GetWindows().GetResources()
+		if resources != nil {
+			if resources.CpuShares > 0 || resources.CpuCount > 0 || resources.CpuMaximum > 0 {
```

Comment:
Why do we have this when each condition is repeated below?

Reply:
This was a limitation in the platform but it looks like our new OCI package here: https://github.com/microsoft/hcsshim/blob/master/internal/oci/uvm.go#L151 properly handles this. So I simplified as you requested.